NGINX Plus R33 support (#2760)
nginxinc/nginx-gateway-fabric: site/content/installation/nginx-plus-jwt.md

---

> title: "NGINX Plus Image and JWT Requirement"

Reviewer: ```suggestion
title: "NGINX Plus and JWT requirement"
```

Author: The reason I had Image in here is because of the need to create the Docker Registry Secret to pull the NGINX Plus docker image, which is an extra step before creating the JWT token Secret.

Reviewer: In context it makes sense, but the page title itself isn't what provides the context, it's the instructions.

We try to prioritize conciseness anywhere we can get it: there might be a better title for the document overall TBH.

It's a super minor LOGAF concern though: I can think about it more in the future.

weight: 300
toc: true
docs: "DOCS-000"
---

## Overview

> NGINX Gateway Fabric with NGINX Plus requires a valid JSON Web Token (JWT) to download the container image from the F5 registry. In addition, starting with version 1.5.0, this JWT token is also required to run NGINX Plus.

Reviewer: ```suggestion
NGINX Gateway Fabric with NGINX Plus requires a valid JSON Web Token (JWT) to download the container image from the F5 registry. From version 1.5.0, this JWT token is also required to run NGINX Plus.
```

This requirement is part of F5’s broader licensing program and aligns with industry best practices. The JWT will streamline subscription renewals and usage reporting, helping you manage your NGINX Plus subscription more efficiently. The [telemetry](#telemetry) data we collect helps us improve our products and services to better meet your needs.

The JWT is required for validating your subscription and reporting telemetry data. For environments connected to the internet, telemetry is automatically sent to F5’s licensing endpoint. In offline environments, telemetry is routed through [NGINX Instance Manager](https://docs.nginx.com/nginx-instance-manager/). Usage is reported every hour and on startup whenever NGINX is reloaded.

> ## Setting up the JWT

Reviewer: ```suggestion
## Set up the JWT
```

Removing the gerund.

> The JWT needs to be configured before deploying NGINX Gateway Fabric. We'll store the JWT in two Kubernetes Secrets. One will be used for downloading the NGINX Plus container image, and the other for running NGINX Plus.

Reviewer: ```suggestion
The JWT needs to be configured before deploying NGINX Gateway Fabric. The JWT will be stored in two Kubernetes Secrets: one for downloading the NGINX Plus container image, and the other for running NGINX Plus.
```

{{% /tab %}}

{{%tab name="Manifests"%}}

Specify the CA Secret name in the `--usage-report-ca-secret` command-line flag on the `nginx-gateway` container. Specify the client Secret name in the `--usage-report-client-ssl-secret` command-line flag on the `nginx-gateway` container.

You also need to define the proper volume mount to mount the Secrets to the nginx container. Add the following volume to the Deployment:

```yaml
- name: nginx-plus-usage-certs
  projected:
    sources:
    - secret:
        name: nim-ca
    - secret:
        name: nim-client
```

and the following volume mounts to the `nginx` container:

```yaml
- mountPath: /etc/nginx/certs-bootstrap/
  name: nginx-plus-usage-certs
```

Finally, in the `nginx-includes-bootstrap` ConfigMap, add the following lines to the `mgmt` block:

```text
ssl_trusted_certificate /etc/nginx/certs-bootstrap/ca.crt;
ssl_certificate        /etc/nginx/certs-bootstrap/tls.crt;
ssl_certificate_key    /etc/nginx/certs-bootstrap/tls.key;
```

{{% /tab %}}

{{</tabs>}}

<br>

**Once these Secrets are created and configuration options are set, you can now [install NGINX Gateway Fabric]({{< relref "installation/installing-ngf" >}}).**

> ## Installation flags for configuring usage reporting {#flags}

Reviewer: ```suggestion
## Installation flags to configure usage reporting {#flags}
```

- **NGINX version and status**: The version of NGINX Plus running on the instance.
- **Instance UUID**: A unique identifier for each NGINX Plus instance.
- **Traffic data**:
  - **Bytes received from and sent to clients**: HTTP and stream traffic volume between clients and NGINX Plus.
  - **Bytes received from and sent to upstreams**: HTTP and stream traffic volume between NGINX Plus and upstream servers.
  - **Client connections**: The number of accepted client connections (HTTP and stream traffic).
  - **Requests handled**: The total number of HTTP requests processed.
- **NGINX uptime**: The number of reloads and worker connections during uptime.
- **Usage report timestamps**: Start and end times for each usage report.
- **Kubernetes node details**: Information about Kubernetes nodes.

### Security and privacy of reported data

All communication between your NGINX Plus instances, NGINX Instance Manager, and F5’s licensing endpoint (`product.connect.nginx.com`) is protected using **SSL/TLS** encryption.

Only **operational metrics** are reported — no **personally identifiable information (PII)** or **sensitive customer data** is transmitted.

> ## Pulling an image for local use

Reviewer: ```suggestion
## Pull an image for local use
```
